src/webargs/core.py

```diff
@@ -531,41 +446,98 @@ def handle_error(error, req, schema, status_code, headers):
         self.error_callback = func
         return func
 
+    def _handle_invalid_json_error(self, error, req, *args, **kwargs):
+        """Internal hook for overriding treatment of JSONDecodeErrors.
+
+        Invoked by default `load_json` implementation.
+
+        External parsers can just implement their own behavior for load_json ,
+        so this is not part of the public parser API.
+        """
+        raise error
+
+    def load_json(self, req, schema):
+        """Load JSON from a request object or return `missing` if no value can
+        be found.
+        """
+        # NOTE: although this implementation is real/concrete and used by
+        # several of the parsers in webargs, it relies on the internal hooks
+        # `_handle_invalid_json_error` and `_raw_load_json`
+        # these methods are not part of the public API and are used to simplify
+        # code sharing amongst the built-in webargs parsers
+        json_data = self._cache.get("json")
+        if json_data is None:
+            try:
+                json_data = self._raw_load_json(req)
+                if json_data is missing:
+                    return missing
+                self._cache["json"] = json_data
+            except json.JSONDecodeError as e:
+                if e.doc == "":
+                    return missing
+                else:
+                    return self._handle_invalid_json_error(e, req)
+            except UnicodeDecodeError as e:
+                return self._handle_invalid_json_error(e, req)
+
+        return json_data
+
```

Comment:
I missed this in my pass over things today.
If I had to guess, I'd say I was just imitating what came before. I see no reason not to cache `missing` results -- I'll make the change.

Reply:
I've just changed this. One test which explicitly checked `self._cache` needed to be tweaked, but nothing else broke with the change.
It's simpler, but I noted in the commit message that we're changing it in part because we don't have context here for why we _shouldn't_ cache `missing`. So if it turns out, on review, that there's some issue which comes up in webargs 6.x because of it, we won't be worried about reverting the change.